### ISSUE TITLE
Calling .read() on a request returns an empty string

### ISSUE THREAD

**Issue by linuxtim** [urllib3/urllib3#715]
I'm not sure whether this is a bug, or a documentation bug, or not a bug at all, but can anyone tell me why reply ends up as:

``` python
b''
```

?

``` python
import urllib3

http = urllib3.PoolManager(timeout=65)

response = http.request('GET', 'http://google.com', decode_content=True)
reply = response.read()
#reply = response.data  # <- this is non-empty however...
print("reply.read() returned: " + reply.decode('utf-8'))
```
##

**Comment by Lukasa** [comment]
Thanks for this! However, if you looked a little more closely at the issues list you might have seen that the second one down is exactly this: #709.

**Comment by Lukasa** [comment]
Arg, I'm sorry, I misread this!

**Comment by Lukasa** [comment]
This is about documentation. You cannot use `read()` by default, because by default all the content is consumed into `data`. If you want `read()` to work, you need to set `preload_content=True` on the call to `urlopen`. Want to give that a try?

**Comment by shazow** [comment]
Yeaaaaaa, this is sucky. I'm sorry. :(

I consider this issue an extension (or dupe?) of https://github.com/shazow/urllib3/issues/436.

**Comment by linuxtim** [comment]
I suppose I expected the content of the internal buffer to be accessible via io operations when preload_content=True was set.

I'm pretty inexperienced with python and was porting some code from urllib2 to take advantage of urllib3's Keep-Alive support, which I guess is likely to be a common use case (to make matters worse, I was POSTing to a server which would respond to invalid content with an empty response - grr).

```diff
diff --git a/docs/index.rst b/docs/index.rst
index 29e7ad7..a3f79c8 100644
--- a/docs/index.rst
+++ b/docs/index.rst
@@ -97,6 +97,11 @@ like buffering:
     >>> # ... your internet connection fails momentarily ...
     >>> secondpart = b.read()
 
+preload_content=False is necessary to prevent urllib3 consuming the server's
+response automatically for access via HTTPResponse.data.  Thus
+**preload_content=False is typically necessary when :mod:`io` style usage is
+required.**
+
 
 Upgrading & Versioning
 ----------------------
```

**Comment by Lukasa** [comment]
@shazow Are you thinking you'd like to have this change in the documentation in the short term, or would you rather just hold off and change the `preload_content` default?

**Comment by linuxtim** [comment]
FWIW, I can't see any harm in making the doc change now, and reverting it when (if?) you make the other change...

A quick google shows that I'm not the only one surprised by this behaviour:

e.g.

http://stackoverflow.com/questions/23514256/http-request-with-timeout-maximum-size-and-connection-pooling

"urllib3: never got the "read" method working,"

**Comment by wooty13** [comment]
On some sites:
import urllib3
http = urllib3.PoolManager()
response = http.request('GET', 'http://python.org', decode_content=True)
reply = response.data
print(reply.decode('utf-8'))

**Comment by wooty13** [comment]
On some sites:
import urllib3
http = urllib3.PoolManager()
response = http.request('GET', 'http://python.org', decode_content=True)
reply = response.data
print(reply.decode('utf-8'))

**Comment by sethmlarson** [comment]
I believe this was fixed in https://github.com/urllib3/urllib3/pull/2798 and will be available in v2.0.0 (which is pre-released right now!)
